### DISCUSSION TITLE
Saving a request.validate() with array to DB

### DISCUSSION THREAD

**Discussion by ad-on-is** [adonisjs/core#2373] [Help]
I have something like this

```js

const productSchema = schema.create({
	title: schema.string(),
	description: schema.string(),
	price: schema.number(),
	images: schema.array([rules.minLength(1)]).members(schema.string()),
	category_id: schema.number(),
	status: schema.string.optional(),
	updated_at: schema.date.optional(),
})

const productData = await request.validate({ schema: productSchema })

await Product.query().where({ id: request.param('id'), userId: auth.user.id }).update(productData)
```
## Problem
This gives me the following error: ` invalid input syntax for type json` since  `productData.images` has to have `JSON.stringify()` applied.

However, when I attempt to do `productData.images = JSON.stringify(productData.images)` , it gives me the error:  `Type 'string' is not assignable to type 'string[]'`

## Dirty solution
Right now, I'm doing something like this to make it work

```js
const productData = JSON.parse(JSON.stringify(await request.validate({ schema: productSchema })))
productData.images = JSON.stringify(productData.images)
await Product.query().where({ id: request.param('id'), userId: auth.user.id }).update(productData)
```

Is there a cleaner solution for this?

**Comment by thetutlage** [comment]
May I know why the images array values are a string? I believe, it will be better to rely on object directly.

```ts
const productSchema = schema.create({
  images: schema
    .array([rules.minLength(1)])
    .members(schema.object().members({
      // object properties goes here
    }))
})
```

**Reply by ad-on-is** [reply]
>May I know why the images array values are a string? I believe, it will be better to rely on object directly.

Out of convenience, I guess :-) Since the client app stores the images **directly** on Firebase Cloud Storage. So, I just store the **filenames** as an array in the DB.

So you mean, something like this would be better, even if it's just the filename?

```js
const productSchema = schema.create({
  images: schema
    .array([rules.minLength(1)])
    .members(schema.object().members({
      fileName: schema.string()
    }))
})
```

Also, would this automatically be saved to the DB, without using `JSON.stringify()`
